File: src/main/java/org/wecancodeit/fleettracker/models/FuelPurchases.java
package org.wecancodeit.fleettracker.models;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class FuelPurchases {

	@Id
	@GeneratedValue
	private Long id;
	private String date;
	private String ticketCheckNumber;
	private String truckNumber;
	private String truckStop;
	private String city;
	private String state;
	private Float gallonsQty;
	// gallonsCost = PUR $AMT on settlement
	private Float gallonsCost;
	private Float authChgbkArrears;
	private Float authChgbkRefund;
	private Float authChgbkNetAmt;

	public FuelPurchases() {

	}

	public FuelPurchases(String date, String ticketCheckNumber, String truckNumber, String truckStop, String city,
			String state, Float gallonsQty, Float gallonsCost, Float authChgbkArrears, Float authChgbkRefund,
			Float authChgbkNetAmt, Float repairMiscAmount) {
		super();
		this.date = date;
		this.ticketCheckNumber = ticketCheckNumber;
		this.truckNumber = truckNumber;
		this.truckStop = truckStop;
		this.city = city;
		this.state = state;
		this.gallonsQty = gallonsQty;
		this.gallonsCost = gallonsCost;
		this.authChgbkArrears = authChgbkArrears;
		this.authChgbkRefund = authChgbkRefund;
		this.authChgbkNetAmt = authChgbkNetAmt;
		this.repairMiscAmount = repairMiscAmount;
	}

	public Long getId() {
		return id;
	}

	public String getDate() {
		return date;
	}

	public String getTicketCheckNumber() {
		return ticketCheckNumber;
	}

	public String getTruckNumber() {
		return truckNumber;
	}

	public String getTruckStop() {
		return truckStop;
	}

	public String getCity() {
		return city;
	}

	public String getState() {
		return state;
	}

	public Float getGallonsQty() {
		return gallonsQty;
	}

	public Float getGallonsCost() {
		return gallonsCost;
	}

	public Float getAuthChgbkArrears() {
		return authChgbkArrears;
	}

	public Float getAuthChgbkRefund() {
		return authChgbkRefund;
	}

	public Float getAuthChgbkNetAmt() {
		return authChgbkNetAmt;
	}

	public Float getRepairMiscAmount() {
		return repairMiscAmount;
	}

	private Float repairMiscAmount;

}

File: fleet-tracker/src/components/FuelPurchases.js
import React, { Component } from 'react';

class FuelPurchases extends Component {

constructor (props) {
	super(props);
}


    render() {
        return (<div>
                <h3>Fuel Purchases</h3>

                <table>
                    <tr>
                        <th>Purchase Date</th>
                        <th>Ticket Check #</th>
                        <th>Truck #</th>
                        <th>Truck Stop</th>
                        <th>City</th>
                        <th>State</th>
                        <th>Gallons QTY</th>
                        <th>Gallons Cost</th>
                        <th>Auth Chgbk Arrears</th>
                        <th>Auth Chgbk Refund</th>
                        <th>Auth Chgbk Net AMT</th>
                        <th>Repair Misc. AMT</th>
                    </tr>

                    {this.props.fuelPurchases.map((fuelPurchase, index) =>
                        <tr key={index}>

                            <td>{fuelPurchase.date}</td>
                            <td>{fuelPurchase.ticketCheckNumber}</td>
                            <td>{fuelPurchase.truckNumber}</td>
                            <td>{fuelPurchase.truckStop}</td>
                            <td>{fuelPurchase.city}</td>
                            <td>{fuelPurchase.state}</td>
                            <td>{fuelPurchase.gallonsQty}</td>
                            <td>{fuelPurchase.gallonsCost}</td>
                            <td>{fuelPurchase.authChgbkArrears}</td>
                            <td>{fuelPurchase.authChgbkRefund}</td>
                            <td>{fuelPurchase.authChgbkNetAmt}</td>
                            <td>{fuelPurchase.repairMiscAmount}</td>
                        </tr>
                    )}
                </table>
            </div>);
    }
}

export default FuelPurchases;